ne/engine.go

```diff
@@ -0,0 +1,229 @@
+package ne
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net/url"
+	"strconv"
+	"sync"
+	"time"
+)
+
+// NotificationStep is one step of a notification process
+type NotificationStep interface {
+	NotifyMethod() string
+	Frequency() time.Duration
+	Until() time.Duration
+}
+
+// NotificationRequest is an interface consumed by the notification handler. It
+// abstracts the differences between people and teams away.
```

Comment:
Thanks. Sadly I could not use the field names from the `NotificationStep` struct since Go does not allow fields and methods with the same name.